### ISSUE TITLE
 msg="containerd: get exit status" error="containerd: process has not exited"

### ISSUE THREAD

**Issue by pineking** [moby/moby#34814]
**Description**

**Steps to reproduce the issue:**
1.
2.
3.

**Describe the results you received:**
Sep 11 17:35:33 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:35:33.729062293+08:00" level=error msg="containerd: get exit status" error="cont
ainerd: process has not exited" id=cf5c86dae68bcc4144fcb4dc7410c533b387b2d7760d6d231b64eb5962a3090f pid=78c4dbc8b091bf62912479909405bb7175a54f484bce5
e60d9ba7dfe0e5b4a69 systemPid=26750
Sep 11 17:37:44 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:37:44.903844066+08:00" level=error msg="containerd: start container" error="cont
ainerd: container did not start before the specified timeout" id=b952194877f79126be707b70a7c30429026cbd8349544f4bab64b72180bed024
Sep 11 17:37:44 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:37:44.904165127+08:00" level=error msg="Create container failed with error: cont
ainerd: container did not start before the specified timeout"
Sep 11 17:37:45 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:37:45.445962357+08:00" level=error msg="Handler for POST /v1.30/containers/b952194877f79126be707b70a7c30429026cbd8349544f4bab64b72180bed024/start returned error: containerd: container did not start before the specified timeout"
Sep 11 17:39:46 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:39:46.141939402+08:00" level=error msg="containerd: start container" error="containerd: container did not start before the specified timeout" id=a900616c99b9af4039ba92cd628c321aa3433d3ef8ca4348c06101ee2eaa146f
Sep 11 17:39:46 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:39:46.142569976+08:00" level=error msg="Create container failed with error: containerd: container did not start before the specified timeout"
Sep 11 17:39:46 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:39:46.695038302+08:00" level=error msg="Handler for POST /v1.30/containers/a900616c99b9af4039ba92cd628c321aa3433d3ef8ca4348c06101ee2eaa146f/start returned error: containerd: container did not start before the specified timeout"
Sep 11 17:39:46 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:39:46.695866951+08:00" level=error msg="Handler for GET /v1.30/containers/json returned error: write unix /var/run/docker.sock->@: write: broken pipe"
Sep 11 17:39:46 24-8a-07-c2-2e-21 dockerd[2435]: http: multiple response.WriteHeader calls
Sep 11 17:41:48 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:41:48.175119309+08:00" level=error msg="containerd: start container" error="containerd: container did not start before the specified timeout" id=54170845d50d5b273361459f00b9ac4f88e9960c628f144893bdf30684cb2525
Sep 11 17:41:48 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:41:48.175468566+08:00" level=error msg="Create container failed with error: containerd: container did not start before the specified timeout"
Sep 11 17:41:48 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:41:48.716103242+08:00" level=error msg="Handler for POST /v1.30/containers/54170845d50d5b273361459f00b9ac4f88e9960c628f144893bdf30684cb2525/start returned error: containerd: container did not start before the specified timeout"
Sep 11 17:41:48 24-8a-07-c2-2e-21 dockerd[2435]: http: multiple response.WriteHeader calls
Sep 11 17:41:48 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:41:48.716647518+08:00" level=error msg="Handler for GET /v1.30/containers/json returned error: write unix /var/run/docker.sock->@: write: broken pipe"
Sep 11 17:44:16 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:44:16.034997741+08:00" level=error msg="containerd: start container" error="containerd: container did not start before the specified timeout" id=1134d17ffd949164a86dd45ca180cb3ae32a7df225795c116850264465619e3e
Sep 11 17:44:16 24-8a-07-c2-2e-21 dockerd[2435]: time="2017-09-11T17:44:16.035507023+08:00" level=error msg="Create container failed with error: containerd: container did not start before the specified timeout"

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:20:36 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:21:56 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 14
 Running: 12
 Paused: 0
 Stopped: 2
Images: 29
Server Version: 17.06.0-ce
Storage Driver: overlay2
 Backing Filesystem: xfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.4.77-1.el7.elrepo.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 56
Total Memory: 251.3GiB
Name: 24-8a-07-c2-2e-21
ID: CPPB:BR7P:DEKY:FZU6:TID2:ZMDC:NQDY:ZRXG:SOIN:XLID:N623:LCU6
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
physical

**Comment by thaJeztah** [comment]
Are you having this issue with every container you start? Also; was this a fresh installation, or an upgrade from a previous version of docker?

Looks like your Docker version is not fully updated; I recommend updating to Docker 17.06.2 as well, which contains various fixes

**Comment by pineking** [comment]
@thaJeztah the previous docker version is 1.13, I uninstalled it and updated the version to 17.06.0 some days ago. 

this issue does not occur when every container creating, it's occasional, perhaps the reason is a huge memory allocated or other heavy loads on the host. since I found khugepaged process at 100% CPU from "top" command output. that caused the docker is unresponsive and timeout. the docker can work normally after a few minutes or so.

I will close this issue and try to update to 17.06.2 to check whether the issue still exists.
